ihmc-high-level-behaviors/src/main/java/us/ihmc/PerceptionAndAutonomyProcess.java

```diff
@@ -567,6 +594,51 @@ public void updatePlanarRegions()
       }
    }
 
+   public void updateHeightMap()
+   {
+      if (realsenseDepthImage != null && realsenseDepthImage.isAvailable() && heightMapDemandNode.isDemanded())
+      {
+         RawImage latestRealsenseDepthImage = realsenseDepthImage.get();
+
+         if (rapidHeightMapExtractor == null)
+         {
+            rapidHeightMapExtractor = new RapidHeightMapExtractor(openCLManager, null); // TODO: Fix up for resetting to feet Z
+            rapidHeightMapExtractor.setDepthIntrinsics(latestRealsenseDepthImage.getIntrinsicsCopy());
+            rapidHeightMapExtractor.create(realsenseDepthImage, 1); // TODO: Can't use same image for whole run?
+         }
+
+         ReferenceFrame d455SensorFrame = realsenseFrameSupplier.get(); // TODO: Can we do this in this thread?
+         ReferenceFrame d455ZUpSensorFrame = realsenseZUpFrameSupplier.get(); // TODO: Can we do this in this thread?
```

Comment:
@TomaszTB @ds58 Is this thread safe? Looks like we do it in other places, seems sketch but maybe it's okay?

Reply:
I don't immediately see the problem